compiler/core/src/swift/swiftConstraint.re

```diff
@@ -380,22 +434,175 @@ let setUpFunction =
       SwiftDocument.joinGroups(
         Empty,
         [
-          root |> Layer.flatmap(translatesAutoresizingMask),
-          constraints |> List.map(defineConstraint),
-          constraints
-          |> List.filter(def => Constraint.getPriority(def) == Low)
+          rootLayer |> Layer.flatmap(translatesAutoresizingMask),
+          allConstraints |> List.map(defineConstraint),
+          /* Priority */
+          allConstraints
+          |> List.filter(const => Constraint.getPriority(const) == Low)
           |> List.map(setConstraintPriority),
-          List.length(constraints) > 0 ? [activateConstraints()] : [],
-          constraints
+          /* Activation */
+          List.length(alwaysConstraints) > 0 ? [activateConstraints()] : [],
+          /* Assign to member variables */
+          conditionalConstraints |> List.map(assignConstraint),
+          alwaysConstraints
           |> List.filter(isDynamic(assignmentsFromLogic))
           |> List.map(assignConstraint),
-          swiftOptions.debugConstraints && List.length(constraints) > 0 ?
+          /* Debug identifiers */
+          swiftOptions.debugConstraints && List.length(allConstraints) > 0 ?
             [
               LineComment("For debugging"),
-              ...constraints |> List.map(assignConstraintIdentifier),
+              ...allConstraints |> List.map(assignConstraintIdentifier),
             ] :
             [],
         ],
       ),
   });
+};
+
+/* private func conditionalConstraints() -> [NSLayoutConstraint] {
```

Comment:
Did you still want this?

Reply:
I think it's kind of convenient but maybe I'll clean it up a little and add some notes